### PR TITLE
use real cache

### PR DESCRIPTION
**Pull request by grosser**
@ggrossman @anamartinez 

this shows some of the bugs gary found ... so let's merge his and then rebase this to see if all got fixed

### CODE REVIEW

File: spec/large_object_store_spec.rb [hunk 1]
```diff
@@ -65,7 +33,7 @@ def delete(key)
   it "cannot read corrupted objects" do
     store.write("a", ["a"*10_000_000]).should == true
     store.store.write("a_4", nil)
-    store.read("a").should == nil
+    store.read("a").class.should == NilClass
```
Comment (anamartinez) [inline spec/large_object_store_spec.rb]: What's the difference?

Comment (grosser) [inline spec/large_object_store_spec.rb]: it won't print 10 million characters if it fails :D

Comment (anamartinez) [inline spec/large_object_store_spec.rb]: ahh of course

**Comment by grosser** [comment]
@ggrossman looking good ?

File: lib/large_object_store.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ def write(key, value, options = {})
       pages = (value.size / slice_size.to_f).ceil
 
       if pages == 1
-        @store.write(key(key, 0), value, options)
+        !!@store.write(key(key, 0), value, options)
```
Comment (grosser) [inline lib/large_object_store.rb]: returns a number in dalli store

**Pull request merged**